From 2cbfe32a5311b9fc13a679a42f6726d35538c160 Mon Sep 17 00:00:00 2001
From: Zhikui Ren <zhikui.ren@intel.com>
Date: Fri, 11 Feb 2022 17:50:14 -0800
Subject: [PATCH] eSPI: add common defines used by optional features.

Add eSPI register definitions. These defines are used
by features that can be included in different orders for
different build options.

Signed-off-by: Zhikui Ren <zhikui.ren@intel.com>
---
 arch/arm/mach-aspeed/ast2600/ast-espi.c | 25 ++++++++++++++++++++++---
 1 file changed, 22 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-aspeed/ast2600/ast-espi.c b/arch/arm/mach-aspeed/ast2600/ast-espi.c
index 71604c4a7f..04afdf4f6b 100644
--- a/arch/arm/mach-aspeed/ast2600/ast-espi.c
+++ b/arch/arm/mach-aspeed/ast2600/ast-espi.c
@@ -9,6 +9,9 @@
 #define AST_SCU_BASE			0x1e6e2000
 #define AST_SCU_HW_STRAP2		0x510
 #define SCU_HW_STRAP_ESPI_ENABLED	0x40
+#define SCU_HW_STRAP_SAFS_ENABLED	BIT(7)
+#define AST_SCU_DEBUG_CTRL2		0x0D8
+#define SCU_DISABLE_ESPI_PERIPHERAL	BIT(0) /* 0 to enable peripheral channel */
 
 #define USE_HW_HANDSHAKE		0
 #define DEBUG_ESPI_ENABLED		0
@@ -88,6 +91,8 @@
 #define AST_ESPI_OOB_CHRDY		BIT(4)
 #define AST_ESPI_FLASH_SW_CHRDY		BIT(7)
 #define AST_ESPI_FLASH_SW_READ		BIT(10)
+#define AST_ESPI_FLASH_HW_MODE		BIT(11)
+#define AST_ESPI_FLASH_MODE		GENMASK(11,10)
 
 /* ESPI00C bits (Interrupt Enable) */
 #define AST_ESPI_IEN_HW_RST		BIT(31)
@@ -102,9 +107,15 @@
 #define AST_ESPI_HW_RST			BIT(31)
 
 /* ESPI080 bits */
-#define AST_ESPI_AUTO_ACK_HOST_RST_WARN	BIT(2)
-#define AST_ESPI_AUTO_ACK_OOB_RST_WARN	BIT(1)
-#define AST_ESPI_AUTO_ACK_SUS_WARN	BIT(0)
+#define AST_ESPI_AUTO_ACK_SUS_WARN			BIT(0)
+#define AST_ESPI_AUTO_ACK_OOB_RST_WARN			BIT(1)
+#define AST_ESPI_AUTO_ACK_HOST_RST_WARN			BIT(2)
+#define AST_ESPI_DISABLE_PERIPHERAL_MEMORY_WRITE	BIT(4)
+#define AST_ESPI_ENABLE_WRITE_PROTECTION_ESPI080_BIT4	BIT(5)
+#define AST_ESPI_DISABLE_PERIPHERAL_MEMORY_READ		BIT(6)
+#define AST_ESPI_ENABLE_WRITE_PROTECTION_ESPI080_BIT6	BIT(7)
+#define AST_ESPI_ENABLE_AUTO_ACK_SL_BT_DONE		BIT(8)
+#define AST_ESPI_ENABLE_AUTO_ACK_SL_BT_STATUS	BIT(9)
 
 /* ESPI098 and ESPI11C bits */
 #define AST_ESPI_OOB_RST_WARN		BIT(6)
@@ -123,6 +134,14 @@
 #define SCR0SIO				0x170
 #define IRQ_SRC_ESPI			74 /* IRQ 74 */
 
+/* SPI controller registers */
+#define AST_SPI0_BASE 			0x1e630000
+
+#define SPI_FLASH_CONFIG_REG 		0x000	/* SPI flash config register */
+#define SPI_ENABLE_CE0_WRITE_DEFAULT 	BIT(16)
+#define SPI_CE_CONTROL_REG 		0x004	/* SPI Control register */
+#define SPI_CE0_ADDRESS_MODE_SEL 	BIT(0)
+
 static void espi_handshake_ack(void)
 {
 	/* IRQ only serviced if strapped, so no strap check */
-- 
2.25.1

